xrootd/xrootd_config.go

```diff
@@ -605,6 +607,67 @@ func LaunchXrootdMaintenance(ctx context.Context, server server_structs.XRootDSe
 	)
 }
 
+// The default config has `Xrootd.AuthRefreshInterval: 5m`, which we need to convert
+// to an integer representation of seconds for our XRootD configuration. This hook
+// handles that conversion during unmarshalling, as well as some sanitization of user inputs.
+func authRefreshStrToSecondsHookFunc() mapstructure.DecodeHookFuncType {
+	return func(f reflect.Type, t reflect.Type, data interface{}) (interface{}, error) {
+		// Filter out underlying data we don't want to risk manipulating
+		if t.Kind() != reflect.Struct || f.Kind() != reflect.Map || t.Name() != "XrootdOptions" {
+			return data, nil
+		}
+
+		// Get the value, load as a time.Duration, and then update the value with seconds as an int
+		dataMap, ok := data.(map[string]interface{})
+		if !ok {
+			return nil, errors.New("data is not a map[string]interface{}")
+		}
+
+		durStr, ok := dataMap["authrefreshinterval"].(string)
+		if !ok {
+			return nil, errors.New("authrefreshinterval is not a string")
+		}
+
+		// Sanitize the input to guarantee we have a unit
+		suffixes := []string{"s", "m", "h", "d"}
+		hasSuffix := false
+		for _, suffix := range suffixes {
+			if strings.HasSuffix(durStr, suffix) {
+				hasSuffix = true
+				break
+			}
+		}
+		if !hasSuffix {
+			log.Warningf("'Xrootd.AuthRefreshInterval' does not have a time unit (s, m, h, d). Interpreting as seconds")
+			durStr = durStr + "s"
+		}
+
+		duration, err := time.ParseDuration(durStr)
+		if err != nil {
+			return nil, err
+		}
+
+		if duration < time.Second {
+			log.Warningf("'Xrootd.AuthRefreshInterval' appears as less than a second. Using fallback of 5m")
```

Comment:
I passed `Xrootd.AuthRefreshInterval: 10s` and xrootd refused to start with the following error:

```console
INFO[2024-06-10T20:32:42Z] 240610 20:32:42 27084 acc_a2x: authrefresh value 10 may not be less than 60  daemon=xrootd.origin
```

We should do this check at Pelican level to get more user friendly error message